source/virtualBuffers/__init__.py

```diff
@@ -260,23 +263,30 @@ def _getPlaceholderAttribute(self, attrs, placeholderAttrsKey):
 					return placeholder
 		return None
 
-	def _getFieldsInRange(self,start,end):
+	def _normalizeCommand(self, command: XMLFormatting.CommandsT) -> XMLFormatting.CommandsT:
+		if not isinstance(command, textInfos.FieldCommand):
+			return command  # no need to normalize str or None
+		field = command.field
+		if isinstance(field, textInfos.ControlField):
+			command.field = self._normalizeControlField(field)
+		elif isinstance(field, textInfos.FormatField):
+			command.field = self._normalizeFormatField(field)
+		return command
+
+	def _getFieldsInRange(self, start: int, end: int) -> textInfos.TextInfo.TextWithFieldsT:
 		text=NVDAHelper.VBuf_getTextInRange(self.obj.VBufHandle,start,end,True)
 		if not text:
-			return ""
-		commandList: typing.List[textInfos.FieldCommand] = XMLFormatting.XMLTextParser().parse(text)
-		for command in commandList:
-			if not isinstance(command, textInfos.FieldCommand):
-				continue  # no need to normalize str or None
-
-			field = command.field
-			if isinstance(field, textInfos.ControlField):
-				command.field = self._normalizeControlField(field)
-			elif isinstance(field, textInfos.FormatField):
-				command.field = self._normalizeFormatField(field)
+			return [""]
+		commandList = XMLFormatting.XMLTextParser().parse(text)
+		commandList = list([

```

Comment:
This creates an extra temporary list.
I don't think you need to wrap this in `list()` as you already have a list expression `[x for x in y if x is not None]`.
 Of course if it were parentheses (a generator expression) then you would need to wrap it in `list()`.